### PR TITLE
UNI-22: Update database design in the cloud

### PR DESCRIPTION
**Pull request by derekxu04**

### CODE REVIEW

File: cloud/template.yml [hunk 1]
```diff
@@ -31,3 +31,58 @@ Resources:
...
+    Properties:
+      AttributeDefinitions:
+        - AttributeName: 'id'
+          AttributeType: 'S'
+        - AttributeName: 'faculty'
+          AttributeType: 'S'
+      KeySchema:
+        - AttributeName: 'id'
+          KeyType: 'HASH'
+      ProvisionedThroughput:
```
Comment (Siyer2) [inline cloud/template.yml]: When you set the ProvisionedThoroughput, it means that the billing mode is 'Provisioned'. The 2 billing modes we have are Provisioned or PayPerRequest. Provisioned means you decide on capacity ahead of time. PayPerRequest only charges per request. As you can see by [this](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BillingModeSummary.html) doc, you can see that Provisioned is great for predictable workloads but PAY_PER_REQUEST is great for unpredictable workloads. For all our tables and indexes, I think this fits better. 

What do you think?

File: cloud/template.yml [hunk 2]
```diff
@@ -31,3 +31,58 @@ Resources:
                   - 'dynamodb:Query'
                 Resource: '*'
   # End IAM Roles
+
+  userTable:
```
Comment (Siyer2) [inline cloud/template.yml:36]: Just like I've done for the IAM roles, I think we should have a comment saying Start DynamoDB Tables and one for the end. In the future, we might have to create separate yml files but for now, clearly separating them by comments is fine.

**Review by Siyer2 — CHANGES_REQUESTED** [review]

**Comment by Siyer2** [comment]
Although setting the branchname automatically links it to Linear, I strongly recommend setting a clearer PR name and description. E.g. maybe something like 'UNI-22: Update database design in the cloud'. This will help us if we need to refer back to this PR in the future

File: cloud/template.yml (hunk 1 above)
Comment (derekxu04) [reply cloud/template.yml]: Yep having had a read of some different resources, it does look like PayPerRequest looks like the way to go.

**Review by Siyer2 — APPROVED** [review]
Awesome work! 🫡

**Pull request merged**